project.md

```diff
@@ -5,47 +5,168 @@
+want to put on it. The notes would be colored/shaped differently depending on
+what hand you should play with and where you should hit the drum.
+
+To make it more convenient, I could have a library of common rhythms (such as
+the horse beat, which is a bit of a swing that sounds like a horse galloping)
+that people could click to drop a collection of notes onto a measure, rather
+than doing it one by one. 
+
+Again, I believe this is the right way because musicians are already familiar
+with GUIs like MuseScore and it clearly works well; there isn't really a need to
+break the formula here except to introduce something that doesn't exist yet (the
+taiko sheet music).
+
+Another thought I have in mind, though, is a simple text input where you input
+the rhythm (for example, writing don don don don would create 4 quarter notes)
+and run the program to make it show up on sheet music. This might be more
+convenient for taiko players specifically because that is how they are taught
+the music. 
+
 ### Operation (semantics)
 
 _What might happen when a program runs? How does a program interact with the
 user? What kinds of errors might occur, and how might they be communicated to
 the user?_
 
+When the program runs, it should export the sheet music as a PDF that can then
+be printed. The user can then read off of the sheet music. Another functionality
+that might be explored in the future is playing back what the music would sound
+like from the program directly. 
+
+Some errors might be trying to fit more notes than there is space (in this case,
+we just deny the user and prevent them from clicking until they delete notes to
+make space). 
+
+Because music is very free-form, though, I don't think there's a lot of errors
+they could make aside from going beyond the constraints of the page.
+
 ### Expressiveness
 
 _What should be easy to do in this language? What should be possible, but
 difficult? What should be impossible or very difficult?_
 
+It should be easy to create sheet music geared for taiko. You could possibly
+adjust this to include other drums, but you'd have to readjust perception on
+what different colors/symbols mean. For example, it wouldn't make sense for a
+regular drum to have sheet music that indicates "dons" and "kas". It should be
+impossible to have non-percussion notes represented because we're not dealing
+with keys.
+
 ### Related work
 
 _Are there any other DSLs in this domain? If not, describe how you know there
 aren't and conjecture why not. If so, describe them and provide links. How well
 do they address the need? Are there any particularly admirable qualities of the
 language? Are there parts of the language you think could be improved?_
 
+Given by the fact that taiko sheet music itself is not a thing, there are no
+DSLs specific to taiko either. There do seem to be some music DSLs floating
+around but again, those are not geared towards percussion instruments, let alone
+taiko.
```

Comment:
I found a DSL written for programming digital drum patterns. It's written in Haskell and not graphical at all, but some of the ideas might be interesting to you, especially with composing tracks: https://www.researchgate.net/publication/321018319_A_Domain_Specific_Language_For_Drum_Beat_Programming